# Estruturas de decisão

#### 1. Write a program that asks for two numbers and prints the larger one.

In [ ]:
n1 = int(input("Enter a number"))

n2 = int(input("Enter another number"))

if n1 > n2:
    print("n1 is the bigger number")
else:
    print("n2 is the bigger number")

#### 2. Make a Program that asks for a value and shows on the screen if the value is positive or negative.

In [ ]:
n1 = float(input("Enter a number"))

if n1 >= 0:
    print("the number is positive")
else:
    print("thee number is negative")

#### 3. Make a program that checks a person's marital status. If the letter entered is "C" (Married), "S" (Single), "D" (Divorced), "V" (Widowed) or "O" (other). According to the letter written by the user, your program must write the marital status, example:

User types: C

Your program should respond:
C - Married

In [ ]:
state = str(input("What state the peaplo is: (C : Maried, S : Single, D : Divorced, V : Window, O : Other)"))

if state == 'C':
    print("C - Married")

elif state == 'S'
    print("S - Single")

elif state == 'D':
    print("D - Divorced")

elif state == 'V':
    print("V - Widow")

elif state == 'O':
    print("O - Other")


#### 4. Write a program that checks if the typed e-mail is part of the spam e-mails.

In [2]:
email = str(input("enter your email: "))

if email:
    if ('@'and '.com') not in email:
        print("enter a valid email")
    else:
        pass
else:
    pass

enter your email: vitor@gmail.com


#### 5.Write a program to read two partial grades from a student. The program must calculate the average achieved per student and present:

##### The "Approved" message, if the average achieved is greater than or equal to seven;
##### The "Fail" message if the average is less than seven;
##### The message "Passed with Distinctions", if the average equals ten.

In [ ]:
grade1 = float(input("Enter the first grade of the student: "))
if grade1:
    grade2 = float(input("Enter the second grade of the student: "))
    if grade2:
        medium = (grade1 + grade2) / 2
        
        if medium >= 7:
            print("The student is aproved")
        elif medium <= 10:
            print("the student is aproved with distinction")
        else:
            print("the student is reproved")
    else:
        print("enter a grade")
else:
    print("enter a grade")

#### 6. Make a Program that reads the budget of 3 companies and shows the biggest one.

In [10]:
company1 = float(input("Enter the first company budget: "))
company2 = float(input("Enter the second company budget: "))
company3 = float(input("Enter the third company budget: "))

if company1 > (company2 and company3):
    print("company 1 have the biggest budget")
elif company2 > (company1 and company3):
    print("company 2 have the biggest budget")
elif company3 > (company2 and company1):
    print("company 3 have the biggest budget")
else:
    pass

Enter the first company budget: 1
Enter the second company budget: 2
Enter the third company budget: 3
company 3 have the biggest budget


#### 7. Make a Program that reads three budgets and shows the biggest and smallest of them.

In [9]:
company1 = float(input("Enter the first company budget: "))
company2 = float(input("Enter the second company budget: "))
company3 = float(input("Enter the third company budget: "))

if company1 > company2 > company3:
    print("company 1 have the bigger budget and the company 3 have the smaller")
elif company1 > company3 > company2:
    print("company 1 have the bigger budget and the company 2 have the smaller")
    
elif company2 > company3 > company1:
    print("company 2 have the bigger budget and the company 1 have the smaller")
elif company2 > company1 > company3:
    print("company 2 have the bigger budget and the company 3 have the smaller")

elif company3 > company2 > company1:
    print("company 3 have the bigger budget and the company 1 have the smaller")
else:
    print("company 3 have the bigger budget and the company 2 have the smaller")   

Enter the first company budget: 3
Enter the second company budget: 2
Enter the third company budget: 1
company 1 have the bigger budget and the company 3 have the smaller


#### 8. Make a program that asks the price of three products and tells you which product you should buy, knowing that the decision is always the cheapest.

In [9]:
prod1 = float(input("Enter the first product price: "))
prod2 = float(input("Enter the second product price: "))
prod3 = float(input("Enter the third product price: "))

if prod1 < prod2 and prod3:
    print("you should buy the first product")
elif prod2 < prod1 and prod3:
    print("you should buy the second product")
elif prod3 < prod1 and prod2:
    print("you should buy the third product")

Enter the first product price: 1
Enter the second product price: 2
Enter the third product price: 3
you should buy the first product


#### 9. Write a program that reads three numbers and displays them in descending order.

In [ ]:
n1 = float(input("Enter the first number: "))
n2 = float(input("Enter a second number: "))
n3 = float(input("Enter a third number: "))

if n1 < n2 < n3:
    print(n1, n2, n3)
elif n1 < n3 < n2:
    print(n1, n3, n2)
    
elif n2 < n1 < n3:
    print(n2, n1, n3)
elif n2 < n3 < n1:
    print(n2, n3, n1)

elif n3 < n1 < n2:
    print(n3, n1, n2)
else:
    print(n3, n2, n1)

#### 10. Make a Program that asks what shift you study. Ask to type M-Morning or V-Evening or N-Night. Print the message "Bom Dia!", "Boa Tarde!" or good night!" or "Invalid Value!", as the case may be.

In [ ]:
period = str(input("Enter the period that you school: (M-Morning, V-Evening, N-Night)"))

if period == 'M':
    print("Good morning!")
elif period == 'V':
    print("Good evening!")
elif period == 'N':
    print("Good night!")

#### 11. The Tabajara Organizations decided to give their collaborators a salary increase and hired you to develop the program that will calculate the readjustments. Make a program that receives the salary of an employee and adjusts it according to the following criteria, based on the current salary:

##### salaries up to R\\$ 280.00 (including): 20% increase

##### salaries between R\\$ 280.00 and R\\$ 700.00: 15% increase

##### salaries between R\\$700.00 and R\\$1500.00: 10% increase

##### wages from R\\$ 1500.00 onwards: 5% increase After the increase is made, inform on the screen:

###### the salary before the readjustment;

###### the percentage increase applied;

###### the increase amount;

###### the new salary, after the increase.

###### Obs: Let's not worry so much about formatting numbers (number of decimal places, for example, we'll see that in the next module)

In [14]:
wage = float(input("Enter the wage of the worker: "))
increase = float()

if wage <= 280:
    increase = wage * 0.2
    print("The wage of ${:.2f} had a increase of 20% ({:.2f}) resulting in a new wage of {:.2f}".format(wage, increase, wage + increase))

elif 700 > wage > 280 :
    increase = wage * 0.15
    print("The wage of ${:.2f} had a increase of 15% ({:.2f}) resulting in a new wage of {:.2f}".format(wage, increase, wage + increase))

elif 1500 > wage > 700:
    increase = wage * 0.1
    print("The wage of ${:.2f} had a increase of 10% ({:.2f}) resulting in a new wage of {:.2f}".format(wage, increase, wage + increase))

elif wage > 1500:
    increase = wage * 0.05
    print("The wage of ${:.2f} had a increase of 05% ({:.2f}) resulting in a new wage of {:.2f}".format(wage, increase, wage + increase))

Enter the wage of the worker: 1
The wage of $1.00 had a increase of 20% (0.20) resulting in a new wage of 1.20


#### 12 . Make a program to calculate a payroll, knowing that the discounts are from the Income Tax, which depends on the gross salary (as shown in the table below) and 3% for the Union and that the FGTS corresponds to 11% of the Gross Salary, but it is not deducted (it is the company that deposits). Net Salary corresponds to Gross Salary minus discounts. The program should ask the user for the hourly rate and the number of hours worked in the month.

IR discount:<br>
Gross Salary up to 900 (inclusive) - exempt<br>
Gross Salary up to 1500 (inclusive) - 5% discount<br>
Gross Salary up to 2500 (inclusive) - 10% discount<br>
Gross Salary above 2500 - 20% discount<br>
Print the information on the screen, arranged according to the example below. In the example the hour value is 5 and the hour amount is 220.

Gross Salary: (5 * 220) : R\\$ 1100.00<br>
(-) IR (5%) : R\\$ 55.00<br>
(-) INSS (10%) : R\\$ 110.00<br>
FGTS (11%): R\\$ 121.00<br>
Total discounts: R\\$ 165.00<br>
Net Salary: R\\$ 935.00<br>

###### Obs: Let's not worry so much about formatting numbers (number of decimal places, for example, we'll see that in the next module)

In [8]:
v_hour = float(input("Enter the value per hour : "))
w_hour = int(input("Worked hours: "))
wage = v_hour * w_hour
print(f'Brute wage: (R${v_hour:.2f} * {w_hour:.2f}): R$ {wage:.2f}')

if wage <= 900:
    percent = 0
elif wage <= 1500:
    percent = 0.05
elif wage <= 2500:
    percent = 0.1
else:
    percent = 0.2

ir = percent * wage
inss = 0.1 * wage
fgts = 0.11 * wage
total_descont = ir + inss
final_wage = wage - total_descont
print(f'(-) IR ({percent * 100}%): R$ {ir:.2f}')
print(f'(-) INSS (10%): R$ {inss:.2f}')
print(f'FGTS (11%): R$ {fgts:.2f}')
print(f'Total de descontos: R$ {total_descont:.2f}')
print(f'Salário Líquido: R$ {final_wage:.2f}')
    


Enter the valueper hour : 1
Worked hours: 2
Brute wage: (R$1.00 * 2.00): R$ 2.00
(-) IR (0%): R$ 0.00
(-) INSS (10%): R$ 0.20
FGTS (11%): R$ 0.22
Total de descontos: R$ 0.20
Salário Líquido: R$ 1.80


#### 13. Write a program that reads a number and displays the corresponding day of the week. (1-Sunday, 2-Monday, etc.), if you type another value, an invalid value should appear.

In [24]:
day = int(input("Enter a day of the week in number: "))

if day == 1:
    print("its sunday")

elif day == 2:
    print("its monday")

elif day == 3:
    print("its tuesday")

elif day == 4:
    print("its wednesday")

elif day == 5:
    print("its thursday")

elif day == 6:
    print("its friday")

elif day == 7:
    print("its saturday")
else:
    print("enter a valid day")

Enter a day of the week in number: 1
its sunday


#### 14. Make a program that reads the two partial grades obtained by a student in a course during a semester, and calculates their average. Then show what concept the student had. The attribution of concepts follows the table below:
<pre>
Concept Achievement Average
Between 9.0 and 10.0 A
Between 7.5 and 9.0 B
Between 6.0 and 7.5 C
Between 4.0 and 6.0 D
Between 4.0 and zero E
</pre>

In [ ]:
grade1 = float(input("Enter the first grade of the student: "))
grade2 = float(input("Enter the second grade of the student: "))
medium = (grade1 + grade2) / 2

if medium <= 4:
    print("Class E")

elif 4 > medium <= 6:
    print("Class D")

elif 6 > medium <= 7.5:
    print("Class c")

elif 7.5 > medium <= 9:
    print("Class b")

elif 9 > medium <= 10:
    print("Class a")

#### 15. VYou are building a calendar to track workdays at the request of HR. In this construction, you will need to define which years are leap years and which are not, to assemble the calendar correctly. Make a Program that asks for a number corresponding to a certain year and then informs if this year is a leap year or not.
<pre>
Tip for determining whether a year is a leap year:
- All years that are multiples of 400 are leap years, e.g.: 1600, 2000, 2400, 2800...
- All multiples of 4 are leap years, except if it is a multiple of 100 but not of 400,
eg: 1996, 2000, 2004, 2008, 2012, 2016, 2020, 2024, 2028...
- All other years are not leap years.<br>
ex1: 2004 is a multiple of 4, but it is not a multiple of 100, so it is a leap year.
ex2: 2000 is a multiple of 4, but it is a multiple of 100, but it is also a multiple of 400, so it is a leap year (because every year that is a multiple of 400 is a leap year, regardless of the rest).
ex3: 1900 is a multiple of 4, it is a multiple of 100, but it is not a multiple of 400, so it is not a leap year

</pre>

Tip: remember that: number % 4 is the remainder of the number divided by 4, eg: 10 % 3 = 1 (since 10/3 = 3 and 1 remains)



In [ ]:
year = int(input("Enter a year: "))
if year:
    if (year % 4 == 0 and ano % 100 != 0) or ano % 400 == 0:
        print("It is a leap year")
    else:
        print("It isnt a leap year")    
else:
    print("Enter a year")

#### 16. Make a Program to read three partial grades from a student. The program must calculate the average achieved per student and present:
<pre>
The message "Approved", if the average is greater than or equal to 7, with the respective average reached;
The message "Failed", if the average is less than 7, with the respective average achieved;
The message "Approved with Distinctions", if the average is equal to 10.
</pre>



In [ ]:
grade1 = float(input("Enter the first grade of the student: "))
if grade1:
    grade2 = float(input("Enter the second grade of the student: "))
    if grade2:
        grade3 = float(input("Enter the third grade of the student: "))
        medium = (grade1 + grade2 + grade3) / 3
        
        if medium >= 7:
            print("The student is aproved")
        elif medium <= 10:
            print("the student is aproved with distinction")
        else:
            print("the student is reproved")
    else:
        print("enter a grade")
else:
    print("enter a grade")

#### 17. John, a good man, bought a microcomputer to control the daily income of his work. Every time he brings a weight of fish greater than that established by the fishing regulations of the state of São Paulo (50 kilos), he must pay a fine of R$ 4.00 per excess kilo. John needs you to write a program that reads the weight variable (fish weight) and calculates the excess. Record in the excess variable the amount of kilos beyond the limit and in the fine variable the amount of the fine that João will have to pay. Print the program data with the appropriate messages.



In [ ]:
weight = float(input("Enter the weight of the fish: "))

if weight > 50:
    fine = (weight - 50) * 4
    print("The fish with a weight of {:.2f} have a fine of {:.2f}".format(weight, fine))
else:
    print("The weight of the fish is under the law")

#### 18. Make a Program for an ATM. The program should ask the user the value of the withdrawal and then inform how many bills of each value will be provided. The notes available will be 1, 5, 10, 50 and 100 reais. The minimum amount is R$10 and the maximum is R$600. The program shouldn't worry about the amount of banknotes in the machine.
<pre>
Example 1: To withdraw the amount of 256 reais, the program provides two 100 bills, a 50 bill, a 5 bill and a 1 bill;
Example 2: To withdraw the amount of 399 reais, the program provides three 100 bills, one 50 bill, four 10 bills, one 5 bill and four 1 bill.
</pre>

Tip1: number // 10 will give you the integer part of dividing the number by 10 as an answer.<br>
Tip2: number % 10 will give you the remainder of the number divided by 10.



In [ ]:
money_take = float(input("Enter the money take: (min )"))

n100 = 0
n50 = 0
n10 = 0
n5 = 0
n1 = 0

if 10 <= money_take <= 600:
    print("It will be needed:")
    if money_take > 100:
        n100 = money_take // 100
        print(n100, "notes of 100")
    if money_take % 100 >= 50:
        n50 = (money_take - 100 * n100) // 50
        print(n50, "notes of 50")
    if money_take % 50 >= 10:
        n10 = (money_take - 100 * n100 - 50 * n50) // 10
        print(n10, "notes of 10")
    if money_take % 10 >= 5:
        n5 = (money_take - 100 * n100 - 50 * n50 - 10 * n10) // 5
        print(n5, "notes of 5")
    if money_take % 5 >= 1:
        n1 = (money_take - 100 * n100 - 50 * n50 - 10 * n10 - 5 * n5)
        print(n1, "notes of 1")

#### 19. Make a program that asks a person 5 questions about a crime. The questions are:
<pre>
"Did you call the victim?"
"Were you at the crime scene?"
"Do you live near the victim?"
"Owed to the victim?"
"Have you ever worked with the victim?"
The program should ultimately issue a rating on the person's participation in the crime. If the person responds positively to 2 questions, he must be classified as "Suspect", between 3 and 4 as "Accomplice" and 5 as "Murderer". Otherwise, he will be classified as "Innocent".
</pre>

In [3]:
print("Answer only 'y' and 'n', 'y' for yes and 'n' for no")
i = 0
telephone = str(input("Phoned the victim: "))
if telephone == 'y':
    i = i + 1
local = str(input("Was in the local of the crime: "))
if local == 'y':
    i = i + 1
live = str(input("Live near the victim: "))
if live == 'y':
    i = i + 1
owed = str(input("owed to the victim: "))
if owed == 'y':
    i = i + 1
work = str(input("Worked with the victim: "))
if work == 'y':
    i = i + 1
if i == 2:
    print("Suspect")
elif 3 <= i <= 4:
    print("Accomplice")
elif i == 5:
    print("Assassin")
else:
    print("Inocent")

Answer only 'y' and 'n', 'y' for yes and 'n' for no
Phoned the victim: y
Was in the local of the crime: y
Live near the victim: y
Worked with the victim: y
Assassin


#### 20. A service station is selling fuel with the following discount table:
<pre>
Alcohol:
up to 20 liters, 3% discount per liter
over 20 litres, 5% discount per liter
Gasoline:
up to 20 liters, 4% discount per liter
over 20 litres, 6% discount per liter

Write an algorithm that reads the number of liters sold, the type of fuel (coded as follows: A-alcohol, G-gasoline), calculates and prints the amount to be paid by the customer, knowing that the price of a liter of gasoline is R$ 2.50 the price of a liter of alcohol is R$ 1.90.
</pre>

In [ ]:
fuel = str(input("Enter the type of fuel :(gas OR alc) "))
liter = float(input("Enter the quantity of liter: "))
discont = float()
price = float()

if fuel == 'alc':
    price = liter * 1.90
    if liter <= 20:
        discont = liter * 0.03
        print("The price was ${:.2f}".format(price - discont ))
    else:
        discont = price * 0.05
        print("The price was ${:.2f}".format(price - discont ))
elif fuel == 'gas':
    price = liter * 2.50
    if liter <= 20:
        discont = price * 0.04
        print("The price was ${:.2f}".format(price - discont))
    else:
        discont = price * 0.06
        print("The price was ${:.2f}".format(price - discont))

#### 21. A fruit shop is selling fruit with the following price list:
<pre>
                      Up to 5 kg Over 5 kg
Strawberry R$ 2.50 per kg R$ 2.20 per kg
Apple R$ 1.80 per kg R$ 1.50 per kg
If the customer buys more than 8 kg of fruit or the total purchase price exceeds R$ 25.00, he will also receive a 10% discount on this total. Write an algorithm to read the amount (in Kg) of strawberries and the amount (in Kg) of apples purchased and write the amount to be paid by the customer.
</pre>

In [ ]:
strawberry = float(input("Enter the kilo of strawberry you want to buy: "))
price_berry = float()
if strawberry <= 5:
    price_berry = strawberry * 2.50
elif strawberry > 5:
    price_berry = strawberry * 2.20
    if price_berry > 25 or strawberry > 8:
        price_berry = price_berry - (price_berry * 0.1)
    else:
        pass

apple = float(input("Enter the kilo of apple you want to buy: "))
price_apple = float()
if apple<= 5:
    price_apple= apple * 2.50
elif apple > 5:
    price_apple= apple * 2.20
    if price_apple> 25 or apple > 8:
        price_apple= price_apple - (price_apple * 0.1)
    else:
        pass

print("The price of strawberry was ${:.2f} and the apple was ${:.2f}".format(price_berry, price_apple))

#### 22. The Hypermarket has a meat promotion that is unmissable. Check out:
<pre>
                      Up to 5 kg Over 5 kg
Double Fillet BRL 4.90 per Kg BRL 5.80 per Kg
Rump BRL 5.90 per Kg BRL 6.80 per Kg
Picanha R$ 6.90 per kg R$ 7.80 per kg
To serve all customers, each customer can only take one of the types of meat in the promotion, but there are no limits to the amount of meat per customer. If the purchase is made using the market card, the customer will also receive a 5% discount on the total purchase. Write a program that asks for the type and quantity of meat purchased by the user and generates a tax receipt, containing the purchase information: type and quantity of meat, total price, type of payment, discount amount and amount to be paid.
</pre>



In [ ]:
type_meat = str(input("Enter the type of meat: (filet, rump, steak)"))
kilo = float(input("KILO: "))
price = float()
payment = str(input("Enter the payment method: (card - client card OR normal)"))

if type_meat == 'filet':
    if kilo <= 5:
        price = kilo * 4.90
        if payment == 'card':
            price = price - (price * 0.05)
        else:
            pass
    else:
        price = kilo * 5.80
        if payment == 'card':
            price = price - (price * 0.05)
        else:
            pass

if type_meat == 'rump':
    if kilo <= 5:
        price = kilo * 5.90
        if payment == 'card':
            price = price - (price * 0.05)
        else:
            pass
    else:
        price = kilo * 6.80
        if payment == 'card':
            price = price - (price * 0.05)
        else:
            pass

if type_meat == 'steak':
    if kilo <= 5:
        price = kilo * 6.90
        if payment == 'card':
            price = price - (price * 0.05)
        else:
            pass
    else:
        price = kilo * 7.80
        if payment == 'card':
            price = price - (price * 0.05)
        else:
            pass
        
print("The total price is ${:.f}".format(price))

#### 23. Make a Program for a paint store. The program should ask for the size in square meters of the area to be painted. Consider that the coverage of the paint is 1 liter for every 6 square meters and that the paint is sold in 18 liter cans, which cost R\\$ 80.00, or in 3.6 liter cans, which cost R\\ $25.00

In [1]:
area = float(input("Enter the area that will be painted: "))
liter_paint = area / 6


Enter the area that will be painted: 1


##### Inform the user with the quantities of ink to be purchased and the respective prices in 3 situations.

Hint: remember the operators // and % shown in previous exercises<br>
Tip1: number // 10 will give you the integer part of dividing the number by 10 as an answer.<br>
Tip2: number % 10 will give you the remainder of the number divided by 10.

##### 1. Buy only 18 liter cans: (whole cans only)

In [3]:
if liter_paint % 18 > 0:
    tin = liter_paint // 18 + 1
else:
    tin = liter_paint // 18
price = tin * 80
print("It will be needed {} tins of paint that will cost ${:.2f}".format(tin, price))

108.0


##### 2. Buy only 3.6 liter gallons: (whole gallons only)

In [ ]:
if liter_paint % 3.6 > 0:
    gallon = liter_paint // 3.6 + 1
else:
    gallon = liter_paint // 3.6
price = gallon * 25
print("It will be needed {} tins of paint that will cost ${:.2f}".format(gallon, price))

##### 3. Mix cans and gallons, so that paint waste is less. Add 10% slack and always round up, ie consider full cans.

The cost of the can is 80/18 = 4.44 R\\$/L

The cost of the gallon is 25/3.6 = 6.94 R\\$/L

The can is more economical, so all the whole cans that we can use must be bought in cans. If something is missing to complete, we must evaluate whether it is better to buy cans or gallons. Example:

If we want to buy 90 liters. 5 cans make exactly 90 liters. So we must buy everything in cans.

If we want to buy 95 liters. 5 cans make exactly 90 liters. So we must buy at least 5 cans and assess what is missing, whether these last 5 liters are worth more only in cans or gallons.

For the missing 5 liters we need 2 gallons that cost 50 reais in total. Or a can that costs 80 reais in total. Therefore, in this case it is better to use 2 gallons.

If we want to buy 107 liters. 5 cans make exactly 90 liters. So we must buy at least 5 cans and assess what is missing, whether these last 5 liters are worth more only in cans or gallons.

For the missing 17 liters we need 5 gallons which cost 125 reais in total. Or a can that costs 80 reais in total. Therefore, in this case it is more worthwhile to use a can.

3 gallons cost 75 reais, 4 gallons cost 100 reais. So, if it is possible to complete with up to 3 gallons, choose gallons. Any quantity greater than 3 gallons, cans are used.

In [12]:
liter_off = liter_paint * 1.1
gallon = 0

tin = liter_off // 18


miss_paint = liter_off - 18 * tin

if miss_paint > 0:
    if miss_paint % 3.6 > 0:
        miss_gallon = miss_paint // 3.6 + 1
    else:
        miss_gallon = miss_paint // 3.6
    if miss_gallon <= 3:
        gallon = miss_gallon
    else:
        gallon = 0
        tin = tin + 1

price = tin * 80 + gallon * 25
print(f"Its needed {tin} tins and {gallon} gallons. The total price is ${price}")

São necessários 0.0 latas e 1.0 galões. Custando R$ 25.0 no total
